vowpalwabbit/feature_group.cc

```diff
@@ -84,6 +94,24 @@ void features::concat(const features& other)
 
   if (!other.space_names.empty())
   { space_names.insert(space_names.end(), other.space_names.begin(), other.space_names.end()); }
+
+  // If the back of the current list and the front of the other list have the same hash then merge the extent.
+  size_t offset = 0;
+  if (!namespace_extents.empty() && !other.namespace_extents.empty())
+  {
+    if (namespace_extents.back().hash == other.namespace_extents.front().hash)
```

Comment:
These if statements can be combined to `if (!namespace_extents.empty() && !other.namespace_extents.empty() && !namespace_extents.empty() && !other.namespace_extents.empty()`

Reply:
Good point!